sopel/coretasks.py

```diff
@@ -368,9 +369,15 @@ def handle_isupport(bot, trigger):
     if 'PREFIX' in bot.isupport:
         bot.modeparser.privileges = set(bot.isupport.PREFIX.keys())
 
-    # was CASEMAPPING support status updated?
-    if not casemapping_support and 'CASEMAPPING' in bot.isupport:
-        # Re-create the bot's nick with the proper identifier+casemapping
+    # rebuild nick when CASEMAPPING and/or CHANTYPES are set
+    if any((
+        # was CASEMAPPING support status updated?
+        not casemapping_support and 'CASEMAPPING' in bot.isupport,
+        # was CHANTYPES support status updated?
+        not chantypes_support and 'CHANTYPES' in bot.isupport,
+    )):
+        # these parameters change how the bot makes Identifiers and since
+        # bot.nick is an Identifier, it must be rebuild
```

Comment:
I decided to tackle style in addition to the typo that first drew my eye.

```suggestion
        # these parameters change how the bot makes Identifiers
        # since bot.nick is an Identifier, it must be rebuilt
```

Reply:
Ha, I wasn't sure about that one!